test/arcade/patch/agent/cell/PatchCellCARTTest.java

```diff
@@ -0,0 +1,235 @@
+
+        when(parameters.getDouble("NECROTIC_FRACTION"))
+                .thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getDouble("EXHAU_FRAC")).thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getDouble("SENESCENT_FRACTION"))
+                .thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getDouble("ANERGIC_FRACTION"))
+                .thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getDouble("PROLIFERATIVE_FRACTION"))
+                .thenReturn(randomIntBetween(40, 100) / 100.0);
+        when(parameters.getInt("SELF_RECEPTORS")).thenReturn(randomIntBetween(100, 200));
+        when(parameters.getDouble("SEARCH_ABILITY")).thenReturn(1.0);
+        when(parameters.getDouble("CAR_AFFINITY")).thenReturn(10 * Math.pow(10, -7));
+        when(parameters.getDouble("CAR_ALPHA")).thenReturn(3.0);
+        when(parameters.getDouble("CAR_BETA")).thenReturn(0.01);
+        when(parameters.getDouble("SELF_RECEPTOR_AFFINITY")).thenReturn(7.8E-6);
+        when(parameters.getDouble("SELF_ALPHA")).thenReturn(3.0);
+        when(parameters.getDouble("SELF_BETA")).thenReturn(0.02);
+        when(parameters.getDouble("CONTACT_FRAC")).thenReturn(7.8E-6);
+        when(parameters.getInt("MAX_ANTIGEN_BINDING")).thenReturn(10);
+        when(parameters.getInt("CARS")).thenReturn(50000);
+
+        when(parameters.getInt("APOPTOSIS_AGE")).thenReturn(120960);
+        when(parameters.getInt("MAX_DENSITY")).thenReturn(54);
+
+        patchCellCART = new PatchCellMock(container, location, parameters);
+    }
+
+    @Test
+    public void testSetAntigenFlag() {
+        patchCellCART.setAntigenFlag(PatchEnums.AntigenFlag.BOUND_ANTIGEN);
+        assertEquals(PatchEnums.AntigenFlag.BOUND_ANTIGEN, patchCellCART.getAntigenFlag());
+    }
+
+    @Test
+    public void testGetAntigenFlag() {
+        patchCellCART.setAntigenFlag(PatchEnums.AntigenFlag.BOUND_ANTIGEN);
+        assertEquals(PatchEnums.AntigenFlag.BOUND_ANTIGEN, patchCellCART.getAntigenFlag());
+    }
+
+    @Test
+    public void testBindTargetNoNeighbors() {
+        Simulation sim = mock(Simulation.class);
+        PatchLocation loc = mock(PatchLocation.class);
+        MersenneTwisterFast random = mock(MersenneTwisterFast.class);
+        PatchGrid grid = mock(PatchGrid.class);
+        Bag bag = new Bag();
+
+        when(sim.getGrid()).thenReturn(grid);
+        when(grid.getObjectsAtLocation(loc)).thenReturn(bag);
+        when(loc.getNeighbors()).thenReturn(new ArrayList<Location>());
+
+        PatchCellTissue result = patchCellCART.bindTarget(sim, loc, random);
+        assertNull(result);
+        assertEquals(PatchEnums.AntigenFlag.UNBOUND, patchCellCART.getAntigenFlag());
+    }
+
+    @Test
+    public void testBindTargetWithSelfBinding() {
+        // lots of self antigens, not a lot of car antigens
+        Simulation sim = mock(Simulation.class);
```

Comment:
make a `setupMocks` class that sets up these mocks before every test so you don't need to keep setting them